### PR TITLE
[STORM-1403] consider gc childopts in memory calculation for worker JVM

### PR DESCRIPTION
**Pull request by zhuoliu**

### CODE REVIEW

**Comment by revans2** [comment]
+1

File: storm-core/src/jvm/backtype/storm/scheduler/Cluster.java [hunk 1]
```diff
@@ -480,6 +480,27 @@ private String getStringFromStringList(Object o) {
     * */
     private Double getAssignedMemoryForSlot(Map topConf) {
         Double totalWorkerMemory = 0.0;
+        final Integer TOPOLOGY_WORKER_DEFAULT_MEMORY_ALLOCATION = 768;
+
+        String topologyWorkerGcChildopts = null;
```
Comment (unsleepy22) [inline storm-core/src/jvm/backtype/storm/scheduler/Cluster.java]: unnecessary null assignment, same below

File: storm-core/src/jvm/backtype/storm/scheduler/Cluster.java [hunk 2]
```diff
@@ -480,6 +480,27 @@ private String getStringFromStringList(Object o) {
     * */
     private Double getAssignedMemoryForSlot(Map topConf) {
         Double totalWorkerMemory = 0.0;
+        final Integer TOPOLOGY_WORKER_DEFAULT_MEMORY_ALLOCATION = 768;
```
Comment (unsleepy22) [inline storm-core/src/jvm/backtype/storm/scheduler/Cluster.java]: is it necessary since default worker memory is set in defaults.yaml

Comment (zhuoliu) [inline storm-core/src/jvm/backtype/storm/scheduler/Cluster.java]: In case someone manually sets it to be nil in the configuration.

**Comment by hustfxj** [comment]
+1

File: storm-core/src/jvm/backtype/storm/scheduler/Cluster.java (hunk 1 above)
Comment (zhuoliu) [inline storm-core/src/jvm/backtype/storm/scheduler/Cluster.java]: Do not see any harm of this null assignment, more of a habitual choice. Same below.

**Pull request merged**